osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,186 @@
+auto lastExecute(std::string& time_data) {
+  // If timestamp is zero dont convert to UNIX Time
+  if (time_data == "0000000000000000") {
+    return 1LL;
+  } else {
+    // swap endianess
+    std::reverse(time_data.begin(), time_data.end());
+
+    for (std::size_t i = 0; i < time_data.length(); i += 2) {
+      char temp = time_data[i];
+      time_data[i] = time_data[i + 1];
+      time_data[i + 1] = temp;
+    }
+
+    // Convert string to long long
+    unsigned long long last_run =
+        tryTo<unsigned long long>(time_data, 16).takeOr(0ull);
+    if (last_run == 0ull) {
+      LOG(WARNING) << "Failed to convert string to long long: " << time_data;
+      return 1LL;
+    }
+
+    FILETIME file_time;
+    ULARGE_INTEGER large_time;
+    large_time.QuadPart = last_run;
+    file_time.dwHighDateTime = large_time.HighPart;
+    file_time.dwLowDateTime = large_time.LowPart;
+    auto last_time = filetimeToUnixtime(file_time);
+    return last_time;
+  }
+}
+
+// Get execution count
+int executionNum(const std::string& assist_data) {
+  if (assist_data.length() <= 16) {
+    LOG(WARNING) << "Userassist execution count format is incorrect";
+    return -1;
+  }
+
+  std::string execution_count = assist_data.substr(8, 8);
+
+  // swap endianess
+  std::reverse(execution_count.begin(), execution_count.end());
+
+  for (std::size_t i = 0; i < execution_count.length(); i += 2) {
+    char temp = execution_count[i];
+    execution_count[i] = execution_count[i + 1];
+    execution_count[i + 1] = temp;
+  }
+
+  auto count = tryTo<std::size_t>(execution_count, 16);
+  if (count.isError()) {
+    LOG(WARNING) << "Error getting execution count: " << count.takeError();
+    return -1;
+  }
+  return count.get();
+}
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
+
+  for (const auto& uKey : users) {
+    auto keyType = uKey.find("type");
+    auto keyPath = uKey.find("path");
+    if (keyType == uKey.end() || keyPath == uKey.end()) {
+      continue;
+    }
+
+    std::string fullPath = keyPath->second + kFullRegPath;
+    QueryData user_assist_results;
+    queryKey(fullPath, user_assist_results);
+
+    for (const auto& rKey : user_assist_results) {
+      auto keyType = rKey.find("type");
+      auto keyPath = rKey.find("path");
+      if (keyType == rKey.end() || keyPath == rKey.end()) {
+        continue;
+      }
+      std::string full_path_key = keyPath->second + "\\Count";
+
+      QueryData assist_results;
+      queryKey(full_path_key, assist_results);
+
+      for (const auto& aKey : assist_results) {
+        std::string subkey = aKey.at("path");
+
+        // split reg path by \Count\ to get Key values
+        auto count_key = subkey.find("Count\\");
+        auto value_key = subkey.substr(count_key);
+        auto value_key_reg = value_key.substr(6, std::string::npos);
+
+        std::string decoded_value_key = rotDecode(value_key_reg);
```

Comment:
N.B.: `UserAssist` aren't guaranteed to be ROT-13 encoded. If `NoEncrypt` is a `DWORD` of value `1` under the `UserAssist` key, the values are saved in plain text.

I'm not sure how common that is, though, so it might just be worth adding as a `NOTE` or `TODO` for the future in case someone runs into it.

Reply:
Thats true if NoEncrypt is added to the Userassist key ROT13 would be turned off (though i tried adding that DWORD to my Windows 10 VM and it was still ROT13 encoding the values, but i may have doing something wrong). I think even if ROT13 is turned off, the existing values would still be encoded. So it would be easy to check to see if the NoEncrypt is set to 1. But I think it would be difficult to determine which data is ROT13 encoded and which is not?

Regardless ive never seen that added to any of the Windows systems ive looked at.
I would think if a user was going to change the Userassist data they would probably just turn it off by adding the DWORD NoLog?

I can add NOTE/comment though mentioning the NoEncrypt setting